src/Chunk.cpp

```diff
@@ -479,8 +479,15 @@ bool cChunk::HasBlockEntityAt(Vector3i a_BlockPos)
 
 void cChunk::Stay(bool a_Stay)
 {
-	m_StayCount += (a_Stay ? 1 : -1);
-	ASSERT(m_StayCount >= 0);
+	if (a_Stay)
+	{
+		m_StayCount++;
+	}
+	else
+	{
+		ASSERT(m_StayCount != 0);
+		m_StayCount--;
+	}
```

Comment:
I could replace it with
```
ASSERT(a_Stay || (StayCount > 0));
StayCount = a_Stay ? (StayCount + 1) : (StayCount - 1);
```
but this construction was to match the body of SetAlwaysTicked.

ints won't save us from a malfunctioning plugin as a general rule, and not in this specific case either since CanUnload checks StayCount is exactly zero.

"If it wasn't for the STL" :P
What do you do at your work by the way? Sounds cool